src/WebJobs.Script/Description/DotNet/FunctionMetadataResolver.cs

```diff
@@ -222,8 +223,12 @@ public Assembly ResolveAssembly(string assemblyName)
                 // When loading shared assemblies, load into the load-from context and load assembly dependencies
                 assembly = Assembly.LoadFrom(assemblyPath);
             }
-            else if (TryResolvePrivateAssembly(assemblyName, out assemblyPath) ||
-                _packageAssemblyResolver.TryResolveAssembly(assemblyName, out assemblyPath))
+            else if (TryResolvePrivateAssembly(assemblyName, out assemblyPath))
+            {
+                assembly = Assembly.Load(File.ReadAllBytes(assemblyPath));
```

Comment:
This is the key change. The default behavior of `LoadFile` would prevent us from loading new versions if an assembly had been loaded with the same path. With this, we have more control, giving us the ability to bypass that behavior.

One added benefit of this approach is that we're now just reading the bytes and not keeping a lock on the file, allowing for in place updates without a restart or locking issues.